### PR TITLE
Feature/spacial gravity v2

### LINKED ISSUE #6: 2D Spacial Gravity
- [ ] Circular planets
- [ ] Should implement appropriate scaling with distance
- [ ] Should have a discrete region where its gravity pulls
- [ ] Should indicate to the player the perimeter of that region

### PR DESCRIPTION
**Pull request by fpdotmonkey**
You can verify that it works by going to Window>General>Test Runner, hitting PlayMode, and then Run All.

You can also try out a demo of the thing by opening Assets/Scenes/Tests/testSpacialGravity.unity.  Click anywhere to create a missile that will fly forward and be affected by spacial gravity.

Closes #6

### CODE REVIEW

File: Assets/Scripts/Core/CircleGravity2D.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    otherTransform.position);
+            if (otherLocalPosition != _center)
+            {
+                return (_center - otherLocalPosition).normalized * mass
+                    / Mathf.Pow(Vector2.Distance(_center, otherLocalPosition),
+                                2);
+            }
+            else
+            {
+                return Vector2.zero;
```
Comment (cporter) [inline Assets/Scripts/Core/CircleGravity2D.cs]: If there's only one path to a correct return, you can just movethe `return DEFAULT_VALUE` bit to the outside of all the conditionals and make it the default case.

File: Assets/Scripts/Core/SpacialGravityReceiver.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+[RequireComponent(typeof(Collider2D))]
+public class SpacialGravityReceiver : MonoBehaviour
+{
+    private Dictionary<int, CircleGravity2D> _activeGravityRegions;
+    private Rigidbody2D _rigidbody;
+
+    private Vector2 CurrentAcceleration()
+    {
+        Vector2 acceleration = Vector2.zero;
+        foreach (KeyValuePair<int, CircleGravity2D> gravityRegion
```
Comment (cporter) [inline Assets/Scripts/Core/SpacialGravityReceiver.cs:22]: Doing this as a `Dictionary.Reduce()` might be more clear than the for loop and accumulator.

File: Assets/Scripts/Demos/ShootMissileOnClick.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    private void Update()
+    {
+        if (Input.GetButtonDown("Fire1") && missile != null)
+        {
+            GameObject newMissile = Instantiate(
+                missile,
+                _camera.ScreenToWorldPoint(Input.mousePosition)
+                    - _camera.transform.position.z*Vector3.forward,
+                Quaternion.identity);
+            newMissile.GetComponent<Rigidbody2D>().velocity = 4.0f*Vector2.up;
```
Comment (cporter) [inline Assets/Scripts/Demos/ShootMissileOnClick.cs]: Give your magic numbers a name.

**Review by cporter — APPROVED** [review]
A few nitpicks, but overall the intent and function are both clear.

File: Assets/Scripts/Core/SpacialGravityReceiver.cs (hunk 2 above)
Comment (fpdotmonkey) [reply Assets/Scripts/Core/SpacialGravityReceiver.cs:22]: Doesn't look like C# has a nice way to do that

**Pull request merged**